core/src/main/java/org/apache/calcite/rel/AbstractRelNode.java

```diff
@@ -177,7 +177,8 @@ public void register(RelOptPlanner planner) {
     Util.discard(planner);
   }
 
-  public final String getRelTypeName() {
+  // It is not recommended to override this method, but sub-classes can do it at their own risk.
+  public String getRelTypeName() {
```

Comment:
It would be cool if we can also describe a little what the original purpose of this method too ~

Reply:
I think there is a very complete description in the [javadoc of the interface RelNode](https://github.com/apache/calcite/blob/a4fa05458840cfdd93fb5cba16d102b06197539d/core/src/main/java/org/apache/calcite/rel/RelNode.java#L367); we could copy-paste it in here, but I think it would be redundant and less simple to maintain.

Reply:
The original document is not suitable now because it describes the default behavior, one can override it to a different one.

Reply:
The original javadoc is the one from the interface (RelNode), which is a contract, so I think its documentation is still valid. Overridden implementations must respect this contract, but they can implement it in a different way, e.g. returning a String literal with the class name, instead of using the default mechanism of AbstractRelNode based on getClass().getName()